torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -164,6 +165,31 @@ struct BroadcastOpRecord : RecordFunctor {
   std::vector<int64_t> broadcast_dims_;
 };
 
+struct ViewOpRecord : RecordFunctor {
+  ViewOpRecord(
+      std::vector<size_t> _args,
+      std::vector<size_t> _outputs,
+      std::vector<int64_t>& original_sizes,
+      std::vector<int64_t>& new_sizes)
+      : RecordFunctor(std::move(_args), std::move(_outputs)),
+        original_sizes_(std::move(original_sizes)),
+        new_sizes_(std::move(new_sizes)) {}
+  virtual ~ViewOpRecord() = default;
+
+  void operator()(FusionDefinition& fd) final {
+    auto arg = fd.getFusionState(args.at(0))->template as<TensorView>();
```

Comment:
I think this is where we would want to copy before the view. cc'ing @kevinstephano for a second opinion